RadixWallet/Core/SharedModels/Assets/FiatWorth.swift

```diff
@@ -0,0 +1,159 @@
+// MARK: - FiatWorth
+public struct FiatWorth: Sendable, Hashable {
+	enum Worth: Sendable, Hashable {
+		case known(RETDecimal)
+		case unknown
+	}
+
+	var isVisible: Bool
+	let worth: Worth
+	var currency: FiatCurrency
+
+	static func unknownWorth(isVisible: Bool, currency: FiatCurrency) -> Self {
+		.init(isVisible: isVisible, worth: .unknown, currency: currency)
+	}
+}
+
+extension FiatWorth {
+	public static func + (lhs: FiatWorth, rhs: FiatWorth) -> FiatWorth {
+		.init(
+			isVisible: lhs.isVisible,
+			worth: lhs.worth + rhs.worth,
+			currency: lhs.currency
+		)
+	}
+}
+
+extension FiatWorth.Worth {
+	public static func + (lhs: Self, rhs: Self) -> Self {
+		switch (lhs, rhs) {
+		case let (.known(lhsValue), .known(rhsValue)):
+			.known(lhsValue + rhsValue)
+		default:
+			.unknown
+		}
+	}
+}
+
+extension FiatWorth.Worth {
+	static let zero: Self = .known(.zero)
+
+	var value: RETDecimal? {
+		if case let .known(value) = self {
+			return value
+		}
+		return nil
+	}
+
+	var isUnknown: Bool {
+		if case .unknown = self {
+			return true
+		}
+		return false
+	}
+}
+
+// MARK: - FiatWorth + Comparable
+extension FiatWorth: Comparable {
+	public static func < (
+		lhs: Self,
+		rhs: Self
+	) -> Bool {
+		lhs.worth < rhs.worth
+	}
+}
+
+// MARK: - FiatWorth.Worth + Comparable
+extension FiatWorth.Worth: Comparable {
+	public static func < (
+		lhs: Self,
+		rhs: Self
+	) -> Bool {
+		switch (lhs, rhs) {
+		case let (.known(lhsValue), .known(rhsValue)):
+			lhsValue < rhsValue
+		case (.known, .unknown):
+			false
+		case (.unknown, .known):
+			true
+		case (.unknown, .unknown):
+			false
+		}
+	}
+}
+
+extension FiatWorth {
+	private static let hiddenValue = "• • • •"
+	private static let unknownValue = "—"
+
+	func currencyFormatted(applyCustomFont: Bool = false) -> AttributedString {
+		let formatter = NumberFormatter()
+		formatter.numberStyle = .currency
+		formatter.currencyCode = currency.currencyCode
+
+		let value = worth.value ?? .zero // Zero for the unknown case, just to do to the base formatting
+
+		if value < 1 {
+			formatter.maximumFractionDigits = 10
+		}
+
+		let formattedValue = {
+			guard let double = try? value.asDouble(), let value = formatter.string(for: double) else {
+				// Good enough fallback
+				return "\(currency.currencyCode)\(value.formattedPlain())"
```

Comment:
would be nice with two screenshot, one with "proper formatting" and one with this fallback, for the same amount, to see how they differ in display.